### PR TITLE
Don't show pokemon that are in unlisted boxes in the list of clones.

### PR DESCRIPTION
**Pull request by YaManicKill**
Just need to fix the tests for the list of clones. But it's too late at night for that.

### CODE REVIEW

File: test/controller/box.js [hunk 1]
```diff
@@ -577,11 +577,38 @@ describe('BoxController', function () {
...
+    let box, box2;
     beforeEach(async () => {
       const res = await agent.post('/api/v1/box').send({name: 'Pillbox', visibility: 'listed'});
+      const res2 = await agent.post('/api/v1/box').send({name: 'Pillbox2', visibility: 'listed'});
       expect(res.statusCode).to.equal(201);
+      expect(res2.statusCode).to.equal(201);
       box = res.body;
+      box2 = res2.body;
+    });
+    it('updates pokemon _boxVisibility', async () => {
```
Comment (not-an-aardvark) [inline test/controller/box.js]: This looks good, but I think we should also have a similar test for when something is uploaded directly to an unlisted box.

Comment (YaManicKill) [inline test/controller/box.js]: Good point on both accounts.

**Comment by YaManicKill** [comment]
Alrighty, should be done now.

File: test/controller/pokemon.js [hunk 2]
```diff
@@ -1240,8 +1243,8 @@ describe('PokemonController', () => {
       pkmnList = [];
       for (let i = 0; i < maxMultiUploadSize * 2; i++) {
         const res = await agent.post('/api/v1/pokemon')
-          // (this just creates a good mix of visibilities and boxes)
-          .field('box', i % 2 ? generalPurposeBox : unlistedBox)
+          .field('box', generalPurposeBox)
+          // (this just creates a good mix of visibilities)
```
Comment (YaManicKill) [inline test/controller/pokemon.js]: I decided to keep this clone list check to just private stuff, and do the unlisted stuff later.

File: test/controller/pokemon.js [hunk 3]
```diff
@@ -1312,6 +1316,21 @@ describe('PokemonController', () => {
         }
       });
     });
+    it('filters clone list based on box visibility', async () => {
```
Comment (YaManicKill) [inline test/controller/pokemon.js]: This checks both the clone list filtering them out, and the fact that if we upload to a box that is unlisted, that is done straight away (as mentions in the upper test)

**Comment by not-an-aardvark** [comment]
LGTM, but we should we should make sure we have a working migration script before merging this. (I want `master` to always be deployable).

I think the following script should work, but it would be good if you/someone could double-check that. Feel free to merge this if the script works.

```js
db.box.find().forEach(function(box) { db.pokemon.update({box: box._id}, {$set: {_boxVisibility: box.visibility}}, {multi: true}) })
```

**Comment by Raia** [comment]
Looks like it works, so merging.

**Pull request merged**